tt_metal/hw/inc/blackhole/noc_nonblocking_api.h

```diff
@@ -344,8 +387,17 @@ inline __attribute__((always_inline)) void noc_fast_atomic_increment(
     uint32_t incr,
     uint32_t wrap,
     bool linked,
-    bool posted = false) {
+    bool posted = false,
+    uint32_t atomic_ret_val = 0) {
     while (!noc_cmd_buf_ready(noc, cmd_buf));
+    if constexpr (noc_mode == DM_DYNAMIC_NOC) {
+        uint32_t noc_id_reg = NOC_CMD_BUF_READ_REG(noc, 0, NOC_NODE_ID);
+        uint32_t my_x = noc_id_reg & NOC_NODE_ID_MASK;
+        uint32_t my_y = (noc_id_reg >> NOC_ADDR_NODE_ID_BITS) & NOC_NODE_ID_MASK;
+        uint64_t atomic_ret_addr = NOC_XY_ADDR(my_x, my_y, atomic_ret_val);
```

Comment:
comment about why this needs to be reprogrammed?

Reply:
the atmoic and write use the same cmd buf, so the return address will be overwritten by normal writes. Need to have it reprogrammed so that atomic has a valid return addr.